src/commands/force/source/retrieve.ts

```diff
@@ -131,4 +136,14 @@ export class Retrieve extends SourceCommand {
 
     return formatter.getJson();
   }
+
+  private wantsToRetrieveCustomFields(): boolean {
+    const hasCustomField = this.componentSet.toArray().some((sourceComponent: SourceComponent) => {
```

Comment:
could you use `this.componentSet.has({fullName: ComponentSet.WILDCARD, type: 'CustomField'})` which might be more performant than iterating over all of them twice.

we only need to check this method if we're using `--metadata` or `--manifest` right? Can we only call this method when those params are present?

Reply:
@WillieRuemmele I agree we should only check on `-m` and `-x` but the `has` method on `componentSet` is `undefined`, so not sure that solution will work: 
![image](https://user-images.githubusercontent.com/2845947/134531394-e830583f-86a3-444c-a6af-e8d888503a5b.png)

Reply:
condition changed to `this.componentSet.has`